R/ptype-abbr-full.R

```diff
@@ -32,10 +34,21 @@ vec_ptype_full <- function(x, ...) {
 
 #' @export
 #' @rdname vec_ptype_full
-vec_ptype_abbr <- function(x, ...) {
+vec_ptype_abbr <- function(x, ..., prefix_named = FALSE, suffix_shape = TRUE) {
   if (!missing(...)) {
     ellipsis::check_dots_empty()
   }
+
+  abbr <- vec_ptype_abbr_(x, prefix_named = prefix_named, suffix_shape = suffix_shape)
+
+  return(paste0(
+    if (prefix_named && !is.null(vec_names(x))) "named ",
+    abbr,
+    if (suffix_shape) vec_ptype_shape(x)
+  ))
+  UseMethod("vec_ptype_abbr")
+}
+vec_ptype_abbr_ <- function(x, ..., prefix_named = FALSE, suffix_shape = TRUE) {
```

Comment:
We usually suffix these helpers with `_dispatch()`.

Since we deal with these prefixes and suffixes in the generic, `prefix_named` and `suffix_shape` shouldn't be passed on to methods.